cranelift-codegen/src/ir/function.rs

```diff
@@ -202,6 +221,24 @@ impl Function {
     pub fn encode(&self, inst: ir::Inst, isa: &TargetIsa) -> Result<Encoding, Legalize> {
         isa.encode(&self, &self.dfg[inst], self.dfg.ctrl_typevar(inst))
     }
+
+    /// Gets list of all instructions with frame layout changes.
+    pub fn frame_layout_instrs(&self) -> Box<[&ir::Inst]> {
```

Comment:
As I understand, `SecondaryMap` allocates memory for all Inst in the function. Though we need a few at the start (and end). `HashMap` will be a better choice IMHO.

I changed that to return a `HashSet` instead of `Box` and now it is used in wasmtime.

Reply:
> I changed that to return a HashSet instead of Box and now it is used in wasmtime.

I still don't see the point of this function. So your usage in wasmtime is these snippets:
```rust
let frame_layout_instrs = func.frame_layout_instrs();
...
if !frame_layout_instrs.contains(&inst) {
    continue;
}
let frame_layout_cmds = func.inst_frame_layout_changes(inst).expect("frame_layout_cmds");
```
A `HashSet` is just a `HashMap` with `()` as the value. So you're taking the keys of a `HashMap` to create a `HashSet`, then looking up the keys in the `HashSet` in order to see if the `HashMap` contains them, and then immediately looking up the key in the `HashMap`. I don't see what this is accomplishing compared to just looking it up in the `HashMap` directly.

Reply:
okay let's do just that